mock-caas/src/main/java/com/sequenceiq/caas/grpc/service/MockGroupManagementService.java

```diff
@@ -0,0 +1,118 @@
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+import javax.inject.Inject;
+
+import org.springframework.stereotype.Service;
+
+import com.cloudera.thunderhead.service.usermanagement.UserManagementProto;
+import com.sequenceiq.cloudbreak.auth.altus.Crn;
+
+import io.grpc.stub.StreamObserver;
+
+@Service
+class MockGroupManagementService {
+
+    private static final String ENVIRONMENT_ADMIN_RIGHT = "environments/write";
+
+    @Inject
+    private MockCrnService mockCrnService;
+
+    private final Map<String, Map<String, UserManagementProto.Group>> accountGroups = new ConcurrentHashMap<>();
+
+    void listGroups(UserManagementProto.ListGroupsRequest request, StreamObserver<UserManagementProto.ListGroupsResponse> responseObserver) {
+
+        UserManagementProto.ListGroupsResponse.Builder groupsBuilder = UserManagementProto.ListGroupsResponse.newBuilder();
+        if (request.getGroupNameOrCrnCount() == 0) {
+            if (isNotEmpty(request.getAccountId())) {
+                getOrCreateGroups(request.getAccountId()).stream()
+                        .forEach(groupsBuilder::addGroup);
+            }
+            responseObserver.onNext(groupsBuilder.build());
+        } else {
+            request.getGroupNameOrCrnList().stream()
+                    .map(this::getOrCreateGroup)
+                    .forEach(groupsBuilder::addGroup);
+            responseObserver.onNext(groupsBuilder.build());
+        }
+        responseObserver.onCompleted();
+    }
+
+    List<UserManagementProto.Group> getOrCreateGroups(String accountId) {
+        accountGroups.computeIfAbsent(accountId, this::createVirtualGroups);
+        List<UserManagementProto.Group> groups = new ArrayList<>(accountGroups.get(accountId).values());
+        groups.sort(Comparator.comparing(UserManagementProto.Group::getGroupName));
+        return groups;
+    }
+
+    void getWorkloadAdministrationGroupName(UserManagementProto.GetWorkloadAdministrationGroupNameRequest request,
+            StreamObserver<UserManagementProto.GetWorkloadAdministrationGroupNameResponse> responseObserver) {
+        UserManagementProto.GetWorkloadAdministrationGroupNameResponse.Builder respBuilder =
+                UserManagementProto.GetWorkloadAdministrationGroupNameResponse.getDefaultInstance().toBuilder();
+        respBuilder.setWorkloadAdministrationGroupName(generateVirtualGroupName(request.getRightName()));
+        responseObserver.onNext(respBuilder.build());
+        responseObserver.onCompleted();
+    }
+
+    void setWorkloadAdministrationGroupName(UserManagementProto.SetWorkloadAdministrationGroupNameRequest request,
+            StreamObserver<UserManagementProto.SetWorkloadAdministrationGroupNameResponse> responseObserver) {
+        responseObserver.onCompleted();
```

Comment:
The code above in `GrpcUmsClient.setWorkloadAdministrationGroupName()` seems to depend on the workloadAdministrationGroupName being returned in the response. Thus, I suspect, lines 59--62 of this class (from `getWorkloadAdministrationGroupName()`) should be also copied here to populate the response appropriately. (`EnvironmentCreationService.create()` calls `GrpcUmsClient.setWorkloadAdministrationGroupName()`, so I wonder how the current logic can work right with the mock-caas. 😃 )

Reply:
Right, it doesn't work, if the GET is not invoked.

Reply:
Fixed, even the set method will return the proper group value